File: Unidad 4-Arreglos/Ejercicio2.go
//autor: Jorge Alberto Chi leon
//Suma de dos vectores bidimensionales de mismo tamaño
//Entradas: Tamaño de las filas, columnas y numeros que contienen
package main

import "fmt"

func main() {
	const size = 1000

	var input [size][size]int
	var input2 [size][size]int
	var filas int
	var columnas int
	fmt.Println("Dame las dimensiones de las filas: ")
	fmt.Scan(&filas)
	fmt.Println("Dame las dimensiones de las columnas: ")
	fmt.Scan(&columnas)

	for i := 0; i < filas; i++ {
		for j := 0; j < columnas; j++ {
			fmt.Println("Ingresa el input[", i, "][", j, "]")
			fmt.Scan(&input[i][j])
			fmt.Println("Ingresa el input2[", i, "][", j, "]")
			fmt.Scan(&input2[i][j])
			//sumas
			input2[i][j] += input[i][j]
		}
	}

	for i := 0; i < filas; i++ {
		for j := 0; j < columnas; j++ {
			fmt.Println("La suma: [", i, "][", j, "]:", input2[i][j])
		}
	}

}
/*
  QA: Raul Rivadeneyra
  Le asigne los valores 1,1,1,1,1,1,1,1,1 y 1,1,1,1,1,1,1,1,1 a las matrices de 3x3 y me devolvio 2,2,2,2,2,2,2,2,2.
  Le asigne los valores de 1,2 y 1,3 a las matrices de 1x2 y me devolvio 2,5
  Le asigne los valores 2,4,4,2 y 1,2,1,2 a las matrices de 2x2 y me devolvio 3,6,5,4.
  Por lo tanto es correcto
  Solo que se puede modular
*/

File: Unidad 4-Arreglos/Ejercicio2.c
//autor: Jorge Alberto Chi leon
//Suma de dos vectores bidimensionales de mismo tamaño
//Entradas: Tamaño de las filas, columnas y numeros que contienen
#include <time.h>
#include <stdio.h>
void lectura(long* x);
void asignacionMatriz(long filaMatriz, long columnaMatriz, long matriz[][columnaMatriz]);
void SumaMatriz(long filaMatriz, long columnaMatriz, long matriz[][columnaMatriz],
	long matrizUno[][columnaMatriz], long matrizDos[][columnaMatriz]);
void impresionMatriz(long filaMatriz, long columnaMatriz, long matriz[][columnaMatriz]);
int main()
{
	long sizeFilaMatriz = 0, sizeColumnaMatriz = 0;
	//se leen los tamaños de los vectores
	lectura(&sizeFilaMatriz);
	lectura(&sizeColumnaMatriz);
	//se declaran los 3 vectores
	long matrizA[sizeFilaMatriz][sizeColumnaMatriz];
	long matrizB[sizeFilaMatriz][sizeColumnaMatriz];
	long matrizSuma[sizeFilaMatriz][sizeColumnaMatriz];
	//se asignan e imprimen los 2 vectores a sumar
	asignacionMatriz(sizeFilaMatriz, sizeColumnaMatriz, matrizA);
	printf("Matriz 1:\n");
	impresionMatriz(sizeFilaMatriz, sizeColumnaMatriz, matrizA);
	asignacionMatriz(sizeFilaMatriz, sizeColumnaMatriz, matrizB);
	printf("Matriz 2:\n");
	impresionMatriz(sizeFilaMatriz, sizeColumnaMatriz, matrizB);
	//se suman los vectores e imprimen
	SumaMatriz(sizeFilaMatriz, sizeColumnaMatriz, matrizSuma, matrizA, matrizB);
	printf("Matriz suma:\n");
	impresionMatriz(sizeFilaMatriz, sizeColumnaMatriz, matrizSuma);
	return 0;
}
void lectura(long* x){
	//se leen los datos del tamaño de los vectores
    printf("Dame las dimensiones de las matrices: \n");
    scanf("%d", x);
}
void asignacionMatriz(long filaMatriz, long columnaMatriz, long matriz[][columnaMatriz]){
    for (int i = 0; i < filaMatriz; ++i)
	{
		for (int j = 0; j < columnaMatriz; ++j)
		{
			//se avanza una casilla cada vez que se lee un dato del vector
			printf("Dame el valor de la posicion %ld, %ld: \n", i + 1, j + 1);
			scanf("%ld", &matriz[i][j]);
		}
	}
}
void SumaMatriz(long filaMatriz, long columnaMatriz, long matriz[][columnaMatriz],
	long matrizUno[][columnaMatriz], long matrizDos[][columnaMatriz]){
    for (int i = 0; i < filaMatriz; ++i)
	{
		for (int j = 0; j < columnaMatriz; ++j)
		{
			//se suman las casillas de ambos vectores
			matriz[i][j] = matrizUno[i][j] + matrizDos[i][j];
		}
	}
}
void impresionMatriz(long filaMatriz, long columnaMatriz, long matriz[][columnaMatriz]){
    printf("--------\n");
    for (int i = 0; i < filaMatriz; ++i)
	{
		for (int j = 0; j < columnaMatriz; ++j)
		{
			//se imprimen los datos de manera ordenada del vector
			if (j == columnaMatriz - 1)
			{
				printf("%ld\n", matriz[i][j]);
			} else {
				printf("%ld, ", matriz[i][j]);
			}

		}
	}
	printf("--------\n");
}
